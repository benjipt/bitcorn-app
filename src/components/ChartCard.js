import React, { Component } from 'react'
import { BarChart, Bar, Legend, Tooltip, XAxis, YAxis, ResponsiveContainer } from 'recharts'

export default class ChartCard extends Component {
    render() {
        return (
            <div className="container mt-5 p-4 border-black rounded card-custom">
                <div className="text-center">
                    <h5>Balance History</h5>
                </div>
                <hr></hr>
                <ResponsiveContainer width="100%" height={300}>
                    <BarChart data={ this.props.data }>
                        <XAxis dataKey="date" />
                        <YAxis />
                        <Tooltip />
                        <Legend />
                        <Bar dataKey="amount" fill="#3E92CC" />
                    </BarChart>
                </ResponsiveContainer>
            </div>
        )
    }
}
