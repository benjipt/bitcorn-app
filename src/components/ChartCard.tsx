import { Tooltip, XAxis, YAxis, ResponsiveContainer, LineChart, Line } from 'recharts'

interface ChartCardProps {
  data: []
}

export default function ChartCard({ data }: ChartCardProps) {
    return (
        <div className="container mt-5 p-4 border-black rounded card-custom" data-testid="ChartCard-1">
            <div className="text-center">
                <h5>Balance History</h5>
            </div>
            <hr></hr>
            <ResponsiveContainer width="100%" height={550}>
                <LineChart data={ data } className="linechart-style" >
                    <XAxis dataKey="date" />
                    <YAxis />
                    <Tooltip />
                    <Line type="monotone" dataKey="amount" stroke="#FF3366" strokeWidth={4} />
                </LineChart>
            </ResponsiveContainer>
        </div>
    )
}
