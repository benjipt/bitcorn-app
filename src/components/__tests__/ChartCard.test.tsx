import { describe, it, expect } from 'vitest';
import { render, cleanup, screen } from '@testing-library/react';
import { act } from 'react-test-renderer';
import ChartCard from '@/components/ChartCard';

describe('ChartCard', () => {
  afterEach(cleanup);

  it('should render ChartCard component', () => {
    // Render ChartCard component
    render(<ChartCard data={[]} />);
    // Grabs ChartCard component from render
    const chartCardComponent = screen.getByTestId('ChartCard-1');
    // Tests for expected component
    expect(chartCardComponent).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    let tree;
    act(() => {
      const { asFragment } = render(<ChartCard data={[]} />);
      tree = asFragment();
    });
    expect(tree).toMatchSnapshot();
  });
});
