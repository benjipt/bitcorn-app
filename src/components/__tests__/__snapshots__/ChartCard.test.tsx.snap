// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChartCard > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-5 p-4 border-black rounded card-custom"
    data-testid="ChartCard-1"
  >
    <div
      class="text-center"
    >
      <h5>
        Balance History
      </h5>
    </div>
    <hr />
    <div
      class="recharts-responsive-container"
      style="width: 100%; height: 550px; min-width: 0;"
    />
  </div>
</DocumentFragment>
`;
